contracts/UniswapV3Staker.sol

```diff
@@ -325,11 +326,17 @@ contract UniswapV3Staker is
             incentives[incentiveId].rewardToken != address(0),
             'non-existent incentive'
         );
+        require(
+            stakes[params.tokenId][incentiveId]
+                .currentlyStaked != true,
+            'already staked'
+        );
 
         (, uint160 secondsPerLiquidityInsideX128, ) =
             pool.snapshotCumulativesInside(tickLower, tickUpper);
         stakes[params.tokenId][incentiveId] = Stake(
-            secondsPerLiquidityInsideX128
+            secondsPerLiquidityInsideX128,
```

Comment:
sanity checking.... @moodysalem I cannot depend on this for truthiness since this value could be 0 for an out of range position, right?

Reply:
yes, definitely don't depend on this being non-zero, it can also overflow to 0 every `type(uint32).max` seconds

Reply:
good point. looking forward to solc8 default safemath!

Reply:
sorry, to be clear, `secondsPerLiquidityInsideX128` is intended to overflow (and can be equal to 0 after overflow), and that is not an issue unless you are liquidity mining over a period greater than `2**32` seconds

Reply:
oops, actually I think `numberOfStakes` has an overflow vulnerability rn, which I was looking at as I read your comment, so I thought you were talking about that.

Reply:
if you only increment a value and decrement a value of type >64 bits, you don't really have to worry about overflow. 2**64 deposit txs is not possible